maths/aliquot_sum.nim

```diff
@@ -0,0 +1,29 @@
+## Aliquot sum
+## In number theory, the aliquot sum s(n) of a positive integer n is the sum of
+## all proper divisors of n, that is, all divisors of n other than n itself.
+## https://en.wikipedia.org/wiki/Aliquot_sum
+{.push raises: [].}
+
+runnableExamples:
+  import std/strformat
+  for number in [12, 100]:
+    echo fmt"The sum of all the proper divisors of {number} is {aliquotSum(number)}"
+
+func aliquotSum(number: Positive): Natural {.raises: [ValueError].} =
```

Comment:
```suggestion
  const expected = [16, 117]
  for i, number in [12, 100].pairs():
    let sum = aliquotSum(number)
    assert sum == expected[i]
    echo fmt"The sum of all the proper divisors of {number} is {sum}"

func aliquotSum*(number: Positive): Natural {.raises: [ValueError].} =
```

Module-level `runnableExamples` require the procs to be exported. And since I'm proposing a change, also added a testing logic. If there's a RE block, why not use it for additional testing?

Reply:
RE blocks should not be used for testing, but by expliciting the expected values the example improved.

Reply:
That's just a suggestion, Personally, I don't see anything wrong with it, as long as the code block still fulfils its explanatory functions. Personally, I had some bugs not covered by test cases found by checking in RE blocks occasionally.